### DISCUSSION TITLE
anyone else have location issues in v38?

### DISCUSSION THREAD

**Discussion by timothywashere** [streetcomplete/StreetComplete#3541] [General]
Just started happening today.
When I launch SC location tracking is fine.
I switch to another app and SC is in the background for a few seconds.
I switch back to SC and my location is frozen, doesn't move. I can scroll the map and use all functionality, just my location dot doesn't move at all
If I close SC and relaunch, location starts tracking again.
Tried restarting phone, same issue.

Anyone else got similar?
Not sure if it's SC or Android or just my phone, will continue experiments tomorrow.

Android 12, Google pixel 3, UK

**Comment by matkoniecz** [comment]
Had something like this, assumed to be random Xiaomi stupidity (reminder to self: never buy their phone again).

**Reply by timothywashere** [reply]
> Had something like this, assumed to be random Xiaomi stupidity (reminder to self: never buy their phone again).

My last phone was Umidigi and I had very weird  issues, won't be buying one of them ever again.

**Comment by backerman** [comment]
It wasn't showing my location at all but clearing cache and force-quitting fixed the problem. (Surface Duo 2, Android 11, SC v38.0)

**Reply by timothywashere** [reply]
> It wasn't showing my location at all but clearing cache and force-quitting fixed the problem. (Surface Duo 2, Android 11, SC v38.0)

I also had that yesterday or the day before. Only once though

**Comment by timothywashere** [comment]
Further experimentation today makes me think it isn't caused by switching app, it's actually caused when the screen turns off and I have to unlock it.

**Comment by mnalis** [comment]
I've also noticed issues with `38.0` (Huawei P30 Pro, EMUI 11, Android 10); if the problem was there in `37.x`, it was **much** rarer. 

It also seems to that SC loses location either when phone goes to sleep, or when the app is moved to background so another app starts. I've managed to capture the video (but didn't manage to catch logcat in time). I was switching between OsmAnd (which had perfect GPS signal all the time) and SC (which was locked at previous location and didn't move, but orientation sensor worked) while walking all the time.

Nothing I did in SC helped (clicking UI, solving quest, entering menus, uploading quests, panning, rotating, enabling and disabling GPS follow multiple times etc). Neither did switching to OsmAnd and back. Only when I removed SC from recent apps and restarted it, did it finally find GPS signal again.


https://user-images.githubusercontent.com/156656/143662836-6aceb9c3-ae98-4d29-8a63-df2e3421d97b.mp4

Perhaps this should be promoted to issue?

**Comment by timothywashere** [comment]
Thanks for contributing. Since other people have same issue I have opened bug #3543

**Comment by codeinabox** [comment]
Thank you for reporting this bug, I've sadly been experiencing it too. Is there anyway to downgrade the installed version until it's fixed?

**Reply by smichel17** [reply]
Not directly. You can either uninstall and reinstall, or install a debug version alongside the regular version. I've triggered a debug apk to be built here: https://github.com/streetcomplete/StreetComplete/actions/runs/1510402009

**Reply by mnalis** [reply]
@smichel17 Thanks! Do note however that unless a (pre-)release is published for longer storage, that `.apk` from build artifacts [will expire in 24h](https://github.com/streetcomplete/StreetComplete/blob/master/.github/workflows/build-debug-apk.yml#L30)

**Reply by codeinabox** [reply]
> Not directly. You can either uninstall and reinstall, or install a debug version alongside the regular version. I've triggered a debug apk to be built here: https://github.com/streetcomplete/StreetComplete/actions/runs/1510402009

I found the simplest solution was to install version 37.2 of the app using F-Droid

**Reply by westnordost** [reply]
Don't forget to update (back) to v38.1 once it appears on F-Droid though!

**Comment by timothywashere** [comment]
Did the v38 release not have a beta release?
I am set up as a beta tester on google play, so I should notice these issues if my phone gets the beta.

**Reply by westnordost** [reply]
It did, the issue was a regression of a fix for a crash issue in v38.0-beta1 if I remember correctly.
